valora-inc/wallet
# perf: use @react-navigation/native-stack and @react-navigation/native v6

### Description

Moves to @react-navigation/native-stack from @react-navigation/stack for increased performance. Initial testing shows a ~28% memory usage reduction from 350MB to 250MB; however, loading a large amount of transaction in the transaction feed will increase memory usage both on main and in this branch.

Some types errors are ignored when using `@react-navigation/native@6.x.x` and navigators with `5.x.x` versions such as `@react-navigation/drawer` and `@react-navigation/material-top-tabs`: mixing Navigators versions is ok until all v5 packages can be upgraded: [docs](https://reactnavigation.org/docs/upgrading-from-5.x/#note-on-mixing-react-navigation-5-and-react-navigation-6-packages).

#### Code changes
- `import/saga.ts` no longer calls `NavigateClearingStack`. Instead after navigating to the verification education screen the device back is disabled. This was due to issues with the navigation state after restoring on initial app load.
- `popToScreen` is used on `pincodeSet.tsx` to reset the stack during the import flow.

#### Dependency Changes

##### Removed Packages

[@react-navigation/compat](https://www.npmjs.com/package/@react-navigation/compat)
[@react-navigation/core](https://www.npmjs.com/package/@react-navigation/core)
[@react-navigation/stack](https://www.npmjs.com/package/@react-navigation/stack)
    
##### Added Packages
[@react-navigation/native-stack](https://www.npmjs.com/package/@react-navigation/native-stack)
[@react-navigation/elements](https://www.npmjs.com/package/@react-navigation/elements)

##### Upgraded Packages
[@react-navigation/native](https://www.npmjs.com/package/@react-navigation/native) - 5.9.8 -> 6.0.13

### Tested

Tested locally on iOS iPhone 11 15.5 and Android SDK 21, 29 & 31.

### How others should test

#### Devs
- Run the app in profile mode on Android and check the memory usage `yarn dev:android -e mainnet -r -t`.
- Follow the steps in the [profile section of the WALLET.md](https://github.com/valora-inc/wallet/blob/main/WALLET.md#app-profiling-with-android-profiler).
- Check for misaligned buttons, incorrect background colors, or text in headers.
- Check for screens not properly dispatched / pincode entry not dispalyed.

#### QA

- Navigate through onboarding screens and ensure that the proper header is displayed.
- Navigate through the Add & Withdraw flows and ensure the proper header is displayed.
- Navigate through setting items and ensure that the proper header is displayed.
- Navigate through the send and receive flows and ensure that the proper header is displayed.
- Check the QR code scanner header is displayed correctly.
- Check for screens not shown in proper order or pincode entry not displayed.

### Related issues

RET-361

### Backwards compatibility

Yes
